bgpd/bgp_bfd.c

```diff
@@ -53,14 +53,23 @@ static void bfd_session_status_update(struct bfd_session_params *bsp,
 					peer->host);
 			return;
 		}
-		peer->last_reset = PEER_DOWN_BFD_DOWN;
 
-		/* rfc9384 */
-		if (BGP_IS_VALID_STATE_FOR_NOTIF(peer->connection->status))
-			bgp_notify_send(peer->connection, BGP_NOTIFY_CEASE,
-					BGP_NOTIFY_CEASE_BFD_DOWN);
-
-		BGP_EVENT_ADD(peer->connection, BGP_Stop);
+		/* Once the BFD session is UP, and later BGP session is UP,
+		 * BFD notices that peer->su_local changed, and BFD session goes down.
+		 * We should trigger BGP session reset only if BFD session is UP
+		 * earlier than BGP session.
```

Comment:
I am confused by the comment.
Now e.g. frr starts at t=0, bfd is UP at t=10s (show bfd peers), BGP is up at t=20s, is that "BFD session is UP earlier than BGP session"?

Reply:
I think he's talking about two different "up" events here, but I'll let @ton31337 answer.

Reply:
Fixed the comment, it was a bit misleading indeed.